core/src/main/scala/org/apache/spark/TaskContextImpl.scala

```diff
@@ -296,4 +304,39 @@ private[spark] class TaskContextImpl(
   private[spark] override def fetchFailed: Option[FetchFailedException] = _fetchFailedException
 
   private[spark] override def getLocalProperties: Properties = localProperties
+
+
+  override def interruptible(): Boolean = TaskContext.synchronized(_interruptible)
+
+  override def pendingInterrupt(threadToInterrupt: Option[Thread], reason: String): Unit = {
```

Comment:
This API looks weird if `threadToInterrupt` is `None`, as there is nothing to interrupt.

Reply:
When `threadToInterrupt=None`, it would still "interrupt" the task by invoking `TaskContext.markInterrupted()`. It just won't invoke `Thread.interrupt()` on the task thread.